hail/src/main/scala/is/hail/expr/types/physical/PCanonicalBaseStruct.scala

```diff
@@ -242,23 +242,30 @@ class PCanonicalBaseStructSettable(
 
   def settableTuple(): IndexedSeq[Settable[_]] = FastIndexedSeq(a)
 
-  def loadField(cb: EmitCodeBuilder, fieldIdx: Int): IEmitCode = {
+  def loadField(cb: EmitCodeBuilder, i: Int): IEmitCode = {
+    val ptTuple = pt.asInstanceOf[PTuple]
```

Comment:
I think this is the wrong fix. PCBaseStructSettable doesn't seem broken -- the fieldIdx here should just be base struct index, not tuple index.

The fix should be to use the true index here: https://github.com/hail-is/hail/blob/50d0da730381e3afbb6a6ea26ebf705597f5549d/hail/src/main/scala/is/hail/expr/ir/Emit.scala#L730

Looks like Chris's PR that moved a bunch of Emit to IEmit broke this on 4/22.

Reply:
Ah, ok that's a better idea. One sec